src/ex4/client/src/components/todoListComp/TodoListComp.jsx

```diff
@@ -0,0 +1,128 @@
+import "../../App.css";
+import { useEffect, useState, useCallback, useMemo } from "react";
+import Toast from "monday-ui-react-core/dist/Toast";
+import ReactLoading from "react-loading";
+import InputCompConnector from "../inputComp/InputCompConnector";
+import ItemsCompConnector from "../itemsComp/ItemsCompConnector";
+import ButtonComp from "../buttonComp/ButtonComp";
+import clearIcon from "../../images/clearIcon.png";
+import NumOfItemsCompConnector from "../numOfItemsComp/NumOfItemsCompConnector";
+
+function TodoListComp({
+	items,
+	isLoading,
+	lastAction,
+	setAllItemsAction,
+	dispatchClearAll,
+	dispatchIsLoading,
+	dispatchUndoLastAction,
+}) {
+	const [toast, setToast] = useState(false);
+
+	useEffect(() => {
+		setToast(true);
+		let timeOut = setTimeout(() => {
+			setToast(false);
+		}, 5000);
+		return () => clearTimeout(timeOut);
+	}, [lastAction]);
+
+	useEffect(() => {
+		const getAllItems = async () => {
+			await setAllItemsAction();
+			dispatchIsLoading(false);
+		};
+		getAllItems();
+	}, [setAllItemsAction, items.length, dispatchIsLoading]);
+
+	const handleToastButton = useCallback(async () => {
+		await dispatchUndoLastAction(lastAction);
+	}, [dispatchUndoLastAction, lastAction]);
+
+	const toastActions = useMemo(
+		() => [
+			{
+				type: Toast.actionTypes.BUTTON,
+				content: "Undo",
+				onClick: handleToastButton,
+			},
+		],
+		[handleToastButton]
+	);
+
+	const staticComps = useCallback(() => {
+		return (
+			<div className="margin-bottom-element">
+				<h1>Sunday.com</h1>
+				<InputCompConnector />
+				<NumOfItemsCompConnector />
+				{lastAction && (
+					<Toast
+						open={toast}
+						type={Toast.types.POSITIVE}
+						closeable={false}
+						actions={toastActions}
+						onClick={async () => {
+							await handleToastButton();
+						}}
```

Comment:
Same, don't write inline function code on prop and especially async await one

Reply:
It's not asynchronous it's my fault and it's not an inline function I call "handleToastButton" which is an external function that calls redux actions